students/anthony_mckeever/lesson5/assignment_1/mailroom.py

```diff
@@ -0,0 +1,397 @@
+                else:
+                    selection[1]()
+            
+            if selection[1] != print_donors:
+                break
+
+        else:
+            if invalid_opt is None:
+                print("Invalid choice.  Please select from available options.")
+            else:
+                invalid_opt(user_choice)
+                break
+
+
+def print_options(show_opts):
+    """
+    Print the options from a menu.
+
+    :show_opts:     The complete dictionary of the current menu's options.
+    """
+    skip = set([])
+    for key, value in show_opts.items():
+        if key not in skip:
+            opts_string, skippable = get_opts_string(key, show_opts)
+            print("\t" + opts_string + value[0])
+            skip.update(skippable)
+            
+
+def key_from_lower(user_choice, keys):
+    """
+    Return the user's lower case input to a properly cased dictionary key.
+
+    :user_choice:   The choice the user made at a menu.
+    :keys:          The keys from the dictionary the user is choosing from.
+    """
+    lower_keys = {k.lower(): k for k in keys}
+    return lower_keys[user_choice] if user_choice in lower_keys.keys() else None
+
+
+def get_opts_string(key, dictionary):
+    """
+    Return a string for multiple keys where the value the key represents is duplicated in the dictionary and the keys
+    that were combined in a list.
+    
+    :key:   The key in the dictionary to find duplicate values of.
+    """
+    keys = [k for k, v in dictionary.items() if v == dictionary[key]]
+    return ", ".join(keys), keys
+
+
+def print_donors():
+    """
+    Print a list of donors.
+    """
+    print("\nList of Donors:")
+    for donor in donor_dict.keys():
+        print("\t" + donor)
+
+        
+def send_thanks():
+    """
+    Initializes the menu system to prompt the acceptance of a donation and creation of a thank you letter.
+    """
+    menu_system(list_dict, "\n\nLets send thanks!", "\nWho do you want to thank? > ", include_main=True,
+                include_donors=True, invalid_opt=accept_donation)
+
+
+def accept_donation(donor_name):
+    """
+    Accept a new donation from a donor.
+
+    :donor_name:    The donor who donated.
+    """
+    donor = get_donor(donor_name)
+
+    donation_float = 0.0
+    while donation_float <= 0.0:
+        donation = safe_input("How much did they donate? > ")
+
+        if main_or_exit(donation):
+            return
+        
+        try:
+            donation_float = float(donation)
+        except ValueError:
+            donation_float = 0.0
+        finally:            
+            if donation_float <= 0.0:
+                print("Invalid amount.  Try again.")
+
+    donor[0].append(donation_float)
+    email = get_email(donor_name, donation_float)
+
+    print("\n\n----- PLEASE SEND THIS EMAIL TO THE DONOR -----\n\n")
+    print(email)
+    print("\n\n----- PLEASE SEND THIS EMAIL TO THE DONOR -----\n\n")
+
+
+def send__to_all():
```

Comment:
single underscores in the function names.

Reply:
Good catch, I'll remove the double underscore.